src/node_binding.cc

```diff
@@ -415,6 +415,12 @@ inline napi_addon_register_func GetNapiInitializerCallback(DLib* dlib) {
 // cache that's a plain C list or hash table that's shared across contexts?
 void DLOpen(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+
+  if (UNLIKELY(!env->options()->allow_native_addons)) {
+    return THROW_ERR_DLOPEN_FAILED(
```

Comment:
This should probably be its own error code, since it's a different error condition.